extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin-serialization-common/runtime/src/main/java/io/quarkus/resteasy/reactive/kotlin/serialization/common/runtime/JsonConfig.java

```diff
@@ -100,6 +101,28 @@ public class JsonConfig {
     @ConfigItem(defaultValue = "false")
     public boolean useArrayPolymorphism = false;
 
+    /**
+     * Specifies the {@code JsonNamingStrategy} that should be used for all properties in classes for serialization and
+     * deserialization.
+     * This strategy is applied for all entities that have {@code StructureKind.CLASS}.
+     * <p>
+     * <p>
+     * {@code null} by default.
+     * <p>
+     * <p>
+     * This element can be one of two things:
+     * <ol>
+     * <li>the fully qualified class name of a type implements the {@code NamingStrategy} interface and has a no-arg
+     * constructor</li>
+     * <li>a value in the form {@code NamingStrategy.SnakeCase} which refers to built-in values provided by the kotlin
+     * serialization
+     * library itself.
+     * </li>
+     * </ol>
+     */
+    @ConfigItem(name = "naming-strategy")
+    public Optional<String> namingStrategy;
```

Comment:
This doesn't really need to be optional as we are supplying a default value.

Reply:
Surprisingly this *does* work in native mode.  I was surprised, too, because there were no `@RegisterForReflection` annotations.  I thought maybe the `when` statement in the test directly referencing the custom `JsonNamingStrategy` might be tripping some wires but I changed that to a String instead and it still works.  I can see printlns I added in both the JVM and native tests.  My hypothesis at this point is that since `JsonNamingStrategy` shows up in the other parts of the API that graalvm makes a note and tracks all the subclasses it finds of that type.  I honestly don't know otherwise.

As for the `Optional`, does `null` count as a default value?  The impression I got looking at other configs was "no."  It truly is an optional config element and kotlin serialization will happily chug along without it.

Reply:
Interesting about native mode.

As for the Optional thing, we can do it like you have it, just saying that Optional is unnecessary as you have provided a default value